### PR TITLE
BTFS-663: PR required for new metadata implementation

### PR DESCRIPTION
**Pull request by steveyeom**
Also backed out the previous changes for the old metadata implementation.

### CODE REVIEW

File: block.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ type BlockAPI interface {
 	Put(context.Context, io.Reader, ...options.BlockPutOption) (BlockStat, error)
 
 	// Get attempts to resolve the path and return a reader for data in the block
-	Get(context.Context, path.Path) (io.Reader, error)
+	Get(context.Context, path.Path, bool) (io.Reader, error)
```
Comment (taiyangc) [inline block.go:27]: Could you add a description about what this bool flag means?

File: unixfs.go [hunk 2]
```diff
@@ -71,7 +71,7 @@ type UnixfsAPI interface {
 	//
 	// Note that some implementations of this API may apply the specified context
 	// to operations performed on the returned file
-	Get(context.Context, path.Path) (files.Node, error)
+	Get(context.Context, path.Path, bool) (files.Node, error)
```
Comment (taiyangc) [inline unixfs.go:75]: Could you add a description about what this bool flag means?

Comment (steveyeom) [reply unixfs.go:75]: Done

**Review by taiyangc — APPROVED** [review]

File: unixfs.go [hunk 3]
```diff
@@ -70,8 +70,9 @@ type UnixfsAPI interface {
 	// Get returns a read-only handle to a file tree referenced by a path
 	//
 	// Note that some implementations of this API may apply the specified context
-	// to operations performed on the returned file
-	Get(context.Context, path.Path) (files.Node, error)
+	// to operations performed on the returned file.
+	// The bool parameter indicates whether token metadata should be returned.
+	Get(context.Context, path.Path, bool) (files.Node, error)
```
Comment (taiyangc) [inline unixfs.go:75]: @steveyeom For future, we should make this inside a new type `...options.UnixfsGetOption` to make new changes easier. Not urgent just a suggestion.

**Pull request merged**